iotedgedev/buildoptions.py

```diff
@@ -0,0 +1,144 @@
+
+
+def split_kv(kvpair):
+    if '=' in kvpair:
+        return kvpair.split('=', 1)
+    else:
+        return kvpair, ''
+
+
+# Mapping docker build cli command to docker python sdk.
+# Format: cli_key : [sdk_key, parse_func]
+# The commented items are cli build options not supported in python sdk.
+cli_sdk_mapping = {
+    '--add-host': ['extra_hosts', functools.partial(parse_to_dict, split_host)],
+    '--build-arg': ['buildargs', functools.partial(parse_to_dict, split_arg)],
+    '--cache-from': ['cache_from', parse_to_list],
+    # '--cgroup-parent' : ['sdk_key', parse_func],
+    # '--compress' : ['sdk_key', parse_func],
+    # '--cpu-period' : ['sdk_key', parse_func],
+    # '--cpu-quota' : ['sdk_key', parse_func],
+    '--cpu-shares': ['cpushares', parse_container_limits],
+    '-c': ['cpushares', parse_container_limits],
+    '--cpuset-cpus': ['cpusetcpus', parse_container_limits],
+    # '--cpuset-mems' : ['sdk_key', parse_func],
+    # '--disable-content-trust' : ['sdk_key', parse_func],
+    '--file': ['dockerfile', parse_val],
+    '-f': ['dockerfile', parse_val],
+    '--force-rm': ['forcerm', parse_flag],
+    # '--iidfile' : ['sdk_key', parse_func],
+    '--label': ['labels', functools.partial(parse_to_dict, split_kv)],
+    '--memory': ['memory', parse_container_limits],
+    '-m': ['memory', parse_container_limits],
+    '--memory-swap': ['memswap', parse_container_limits],
+    '--network': ['network_mode', parse_val],
+    '--no-cache': ['nocache', parse_flag],
+    '--platform': ['platform', parse_val],
+    '--pull': ['pull', parse_flag],
+    '--quiet': ['quiet', parse_flag],
+    '-q': ['quiet', parse_flag],
+    '--rm': ['rm', parse_flag],
+    # '--security-opt' : ['sdk_key', parse_func],
+    '--shm-size': ['shmsize', parse_val],
+    '--squash': ['squash', parse_flag],
+    # '--stream' : ['sdk_key', parse_func],
+    '--tag': ['tag', parse_val],
+    '-t': ['tag', parse_val],
+    '--target': ['target', parse_val],
+    # '--ulimit' : ['sdk_key', parse_func]
+}
+
+# Class BuildOptions parse docker cli build commands to python sdk dict.
+
+
+class BuildOptions(object):
+    def __init__(self, build_options):
+        self.build_options = build_options
+        self.sdk_options = {}
+
+    def filter_build_options(self, build_options):
+        """Remove build options which will be ignored"""
+        if build_options is None:
+            return None
+
+        filtered_build_options = []
+        for build_option in build_options:
+            build_option = build_option.strip()
+            parsed_option = re.compile(r"\s+").split(build_option)
+            if parsed_option and parsed_option[0] not in ["--rm", "--tag", "-t", "--file", "-f"]:
+                filtered_build_options.append(build_option)
+
+        return filtered_build_options
+
+    def parse_build_options(self):
+        """Parse build options to python SDK"""
+        filtered_build_options = self.filter_build_options(self.build_options)
```

Comment:
Here there is passing of an instance variable to an instance method. Because the class requires and sets build_options in the constructor and parse_build_options doesn't accept an argument, it makes sense for filter_build_options to also not require build_options as an argument and assume operation on the instance variable.